### ISSUE TITLE
Headers don't persist after changing path

### ISSUE THREAD

**Issue by chendo** [htty/htty#136]
I set an Authorization header with `headers-s`, perform the request, which works fine, but when I `cd` to another path, I need to set the header again, which doesn't feel right.

htty v1.5.0

**Comment by njonsson** [comment]
I’ve reproduced this and covered the failed behavior with a spec. It’s rooted in the fact that _htty_ has built-in support for HTTP Basic Authorization but nothing else. It assumes that the _Authorization_ header can be blown away if the URL is being changed and it contains no userinfo (e.g., http://**my_username:my_password@**host).

Feel free to work on [the branch for this issue](https://github.com/htty/htty/tree/path-set-erases-non-basic-authorization-header) and send me a pull request. I will get to it eventually, if not.

Thanks.

**Comment by favrik** [comment]
Hi, I was trying to contribute by fixing this, but I could not reproduce the issue.

Here's how I was trying to reproduce:

```Shell
htty example.com

http://example.com/> 
http://example.com/> get
401  Unauthorized -- 5 headers -- 588-character body

http://example/> headers-set Authorization 'Basic asdfasdfasdfasdfasfas='
http://user:pass@example.com/>
http://user:pass@example.com/> get
 200  OK -- 11 headers* -- 394-character body

http://user:pass@example.com/> cd otherpage.php
http://user:pass@example.com/otherpage.php>
http://user:pass@example.com/otherpage.php> get
 200  OK -- 11 headers* -- 394-character body
```

I didn't need to set the header again.   The Authorization header persists as long as the URL doesn't change.  Thoughts?

**Comment by njonsson** [comment]
Yes, the use case you tried works fine. If the header is an HTTP Basic Authentication header (begins with `Basic `) then it is retained. Other kinds of authentication headers get lost.
